rulebook-spring/src/main/java/com/deliveredtechnologies/rulebook/spring/SpringAwareRuleBookRunner.java

```diff
@@ -0,0 +1,47 @@
+package com.deliveredtechnologies.rulebook.spring;
+
+import com.deliveredtechnologies.rulebook.model.runner.RuleBookRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/**
+ * Runs the POJO Rules that can be Spring aware beans in a specified package as a RuleBook.
+ * <p>
+ * POJO rule classes in the package may or may not have the @Component/@Service/etc. annotation. If they
+ * do, the bean will be fetched and @Autowired variables will be populated. If not, an instance of the
+ * POJO rule will be created and used.
+ */
+public class SpringAwareRuleBookRunner extends RuleBookRunner implements ApplicationContextAware {
+
+  private ApplicationContext _applicationContext;
+
+  private static Logger LOGGER = LoggerFactory.getLogger(SpringAwareRuleBookRunner.class);
+
+  public SpringAwareRuleBookRunner(String rulePackage) {
+    super(rulePackage);
+  }
+
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    this._applicationContext = applicationContext;
+  }
+
+  @Override
+  protected Object getRuleInstance(Class<?> rule) {
+    // For backwards compatibility, if not within a Spring project
```

Comment:
I almost kind of think you should let it blow up if someone wanted to use a SpringAwareRuleBookRunner without using Spring. Kind of serves them right, no? :)
I have no problem with letting that behavior work. But you probably should log a warning or something.

Reply:
Actually at work I'm using a rulebook with around 15 rules only 2 of which require access to bean components. It feels un-friendly to have all 15 rules of them be `@Component`'ized so I strongly feel this behavior is ok since it's the SpringAwareRuleBookRunner that accommodates spring aware rules, not that the ruleBookRunner forces all rules to be spring beans/components. If you still feel otherwise I'll throw an exception. Let me know.